syntax = "proto2";
option java_package = "pl.edu.agh.wiet.sr.banking.generated";

service CurrencyService {
    rpc GetCurrencyRate(CurrencyRequest) returns (stream Rate) {}
}

enum Currency {
    PLN = 0;
    EUR = 1;
    USD = 2;
    CHF = 3;
}

message CurrencyRequest {
    required Currency currencyFrom = 1;
    required Currency currencyTo = 2;
}

message Rate {
    required double value = 2;
}